### DISCUSSION TITLE
Purpose behind optional imports

### DISCUSSION THREAD

**Discussion by jpcenteno80** [Project-MONAI/monai-deploy-app-sdk#162] [Q&A] [answered]
Hi, I am trying to build and operator to convert dicom series to nrrd and was trying to follow the logic behind scripting operators already implemented in the library. In `monai_seg_inference_operator.py`, numpy is imported at the top of the script, and several numpy functions are used in the code. However, torch is imported using `optional_import` and versioned to 1.5. Also several torch functions are used throughout the remainder of the code.

I was wondering why `optional_import` is used in some instances?

Thanks!

**Comment by MMelQin** [comment, marked as answer]
Thanks for the question.

optional_import is used both in MONAI and MONAI Deploy App SDK, so that the package themselves when built/published do not contain those dependent packages, e.g. an application is built to perform classic image processing from a DICOM study, no need to use PyTorch for inference, hence no need to force PyTorch package be in the App SDK.

**Comment by dbericat** [comment]
@jpcenteno80 could you please confirm Ming's answer solved your question to close this discussion? Thanks!
